src/criteria_assessment/query_thresholds.jl

```diff
@@ -173,46 +184,60 @@ function apply_criteria_thresholds(
     res = Raster(criteria_stack.Depth; data=sparse(data), missingval=0)
     return res
 end
-function apply_criteria_thresholds(
-    criteria_stack::T,
-    lookup::DataFrame,
-    ruleset::Vector{CriteriaBounds{Function}}
-)::Raster where {T}
-    # Result store
-    data = falses(size(criteria_stack))
 
-    res_lookup = trues(nrow(lookup))
+"""
+    apply_criteria_lookup(
+        reg_criteria::OldRegionalCriteria,
+        rtype::Symbol,
+        ruleset::Vector{CriteriaBounds{Function}}
+    )
+
+Filter lookup table by applying user defined `ruleset` criteria.
+
+# Arguments
+- `reg_criteria` : OldRegionalCriteria containing valid_rtype lookup table for filtering.
+- `rtype` : Flats or slope category for assessment.
+- `ruleset` : User defined ruleset for upper and lower bounds.
+
+# Returns
+Filtered lookup table containing points that meet all criteria in `ruleset`.
+"""
+function apply_criteria_lookup(
+    reg_criteria::OldRegionalCriteria,
+    rtype::Symbol,
+    ruleset
+)::DataFrame
+    lookup = getfield(reg_criteria, Symbol(:valid_, rtype))
+    lookup.all_crit .= 1
+
     for threshold in ruleset
-        res_lookup .= res_lookup .& threshold.rule(lookup[!, threshold.name])
+        lookup.all_crit = lookup.all_crit .& threshold.rule(lookup[!, threshold.name])
     end
 
-    tmp = lookup[res_lookup, [:lon_idx, :lat_idx]]
-    data[CartesianIndex.(tmp.lon_idx, tmp.lat_idx)] .= true
-
-    res = Raster(criteria_stack.Depth; data=sparse(data), missingval=0)
+    lookup = lookup[BitVector(lookup.all_crit), :]
 
-    return res
+    return lookup
 end
 
 """
     apply_criteria_lookup(
-        reg_criteria::RegionalCriteria,
+        raster_stack::RasterStack,
```

Comment:
Here it specifies `RasterStack` but the arguments list and the implemented function says `OldRegionalCriteria`?